operator/src/test/java/org/keycloak/operator/KeycloakDeploymentE2EIT.java

```diff
@@ -246,4 +251,75 @@ public void testHostnameStrictDisabled() {
         }
     }
 
+    // Reference curl command:
+    // curl --insecure --data "grant_type=password&client_id=admin-cli&username=admin&password=adminPassword" https://localhost:8443/realms/master/protocol/openid-connect/token
+    @Test
+    public void testInitialAdminUser() {
+        try {
+            // Recreating the database to keep this test isolated
+            deleteDB();
+            deployDB();
+            var kc = getDefaultKeycloakDeployment();
+            deployKeycloak(k8sclient, kc, true);
+
+            var decoder = Base64.getDecoder();
+            var service = new KeycloakService(k8sclient, kc);
+            var kcAdminSecret = new KeycloakAdminSecret(k8sclient, kc);
+
+            AtomicReference<String> adminUsername = new AtomicReference<>();
+            AtomicReference<String> adminPassword = new AtomicReference<>();
```

Comment:
Is the `AtomicReference` really necessary? Awaitility doesn't use threads AFAIK.

Reply:
We are using those values in `lambda`s and the IDE screams if they are not `final` or `AtomicRef`s

Reply:
Ah, right. The "effectively final" rule...